File: src/main/kotlin/com/xs0/gqlktx/utils/StringInputStream.java
package com.xs0.gqlktx.utils;

import java.io.InputStream;

public class StringInputStream extends InputStream {
    private final String s;
    private int pos;
    private int len;

    public StringInputStream(String s) {
        this.s = s;
        this.pos = 0;
        this.len = s.length();
    }

    @Override
    public int read() {
        if (pos >= len) {
            return -1;
        } else {
            return s.charAt(pos++) & 255;
        }
    }
}

File: src/main/kotlin/com/xs0/gqlktx/utils/StringInputStream.kt
package com.xs0.gqlktx.utils

import java.io.InputStream

class StringInputStream(private val s: String) : InputStream() {
    private var pos: Int = 0
    private val len: Int = s.length

    override fun read(): Int {
        return if (pos >= len) {
            -1
        } else {
            s[pos++].toInt() and 255
        }
    }
}